internal/gcp/secret-manager.go

```diff
@@ -0,0 +1,176 @@
+// Copyright 2020 - MinIO, Inc. All rights reserved.
+// Use of this source code is governed by the AGPLv3
+// license that can be found in the LICENSE file.
+
+package gcp
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"net/http"
+
+	secretmanager "cloud.google.com/go/secretmanager/apiv1"
+	"github.com/minio/kes"
+	"google.golang.org/api/option"
+	secretmanagerpb "google.golang.org/genproto/googleapis/cloud/secretmanager/v1"
+)
+
+type SecretManager struct {
+	// The GCP project id
+	// Obviously, you need to create GCP project
+	ProjectID string
+
+	// The GCP service account key json file
+	// Follow this documentation to create service account & get the key file
+	// https://cloud.google.com/secret-manager/docs/reference/libraries#setting_up_authentication
+	CredentialsFile string
+
+	// ErrorLog specifies an optional logger for errors
+	// when files cannot be opened, deleted or contain
+	// invalid content.
+	// If nil, logging is done via the log package's
+	// standard logger.
+	ErrorLog *log.Logger
+
+	client *secretmanager.Client
+}
+
+var (
+	errCreateKey = kes.NewError(http.StatusBadGateway, "bad gateway: failed to create key")
+	errGetKey    = kes.NewError(http.StatusBadGateway, "bad gateway: failed to access key")
+	errDeleteKey = kes.NewError(http.StatusBadGateway, "bad gateway: failed to delete key")
+)
+
+// Create stores the given key-value pair at GCP secret manager
+// if and only if it doesn't exists. If such an entry already exists
+// it returns kes.ErrKeyExists.
+//
+// GCP secret manager first need to create the secret itself
+// and then add a secret version which is where the actual key-value pair stored
+// The secret will be stored in encrypted form using AES-256
+func (s *SecretManager) Create(key, value string) error {
+	if s.client == nil {
+		s.logf("gcp: no connection to GCP secret manager: '%s'", s.ProjectID)
+		return errCreateKey
+	}
+
+	existingSecret, _ := s.client.GetSecret(
```

Comment:
We must not ignore the error here. There are to cases to consider. Eventually, GCP returns a well-defined error if a secret is already present in which case we should return `kes.ErrKeyExists`

Also, if another error occurs we have to log that error to the `ErrorLog` / `STDERR`. Otherwise, a user has no way to determine why it's KES-GCP setup does not work. Also, we should return an error with 502 Bad Gateway status code in this case. See e.g. https://github.com/minio/kes/blob/a918c440d5b631a0d2628ae3aec289b6d4cefee1/internal/aws/secrets-manager.go#L91